src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const lengthActions = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const TYPE_MESSAGE = action.type;
+    const secondValue = action.extraData;
+    const keyRemove = action.keysToRemove;
+
+    switch (TYPE_MESSAGE) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy,
+          ...secondValue,
+        };
+
+        lengthActions.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (const key of keyRemove) {
+          if (stateCopy.hasOwnProperty(key)) {
```

Comment:
This check is not necessary, the deletion of a property that does not exist doesn't throw an Error